crates/rover-client/src/error.rs

```diff
@@ -73,8 +74,20 @@ pub enum RoverClientError {
         frontend_url_root: String,
     },
 
+    #[error("Could not connect to {}.",
+        if let Some(url) = .url {
+            url.to_string()
+        } else {
+            "Unknown URL".to_string()
+        }
+    )]
+    CouldNotConnect {
+        source: reqwest::Error,
+        url: Option<Url>
+    },
+
     /// Encountered an error sending the request.
-    #[error("encountered an error while sending a request")]
+    #[error(transparent)]
```

Comment:
i changed this to `transparent` so that it would just print out the whole error message from reqwest. they're a bit verbose but they at least give some info to the user that we were obfuscating previously. i imagine as we see these crop up we can add more of our own error types